src/analysis/scoping_analysis.cpp

```diff
@@ -130,8 +133,18 @@ struct ScopingAnalysis::ScopeNameUsage {
     StrSet got_from_closure;
     StrSet passthrough_accesses; // what names a child scope accesses a name from a parent scope
 
+    // `import *` and `exec` both force the scope to use the NAME lookup
+    bool hasNameForcingSyntax;
```

Comment:
minory, but maybe we could have this as a function if it's just a simple derivation of other fields (`nameForcingNodeImportStar != NULL || nameForcingNodeBareExec != NULL`).